### PR TITLE
Upgrades to syntax and types to prepare for scala 2.13 upgrade

### PR DESCRIPTION
**Pull request by andrew-nowak**
## What does this change?

Lots of updates to deprecated syntax and classes involved in the Scala 2.13 upgrade. Go commit-by-commit for related changes with (very brief!) descriptions of what's changed

## Tested? Documented?
- [ ] locally by committer
- [ ] locally by Guardian reviewer
- [ ] on the Guardian's TEST environment
- [ ] relevant documentation added or amended (if needed)

### CODE REVIEW

File: common-lib/src/main/scala/com/gu/mediaservice/lib/DateTimeUtils.scala [hunk 1]
```diff
@@ -20,7 +20,7 @@ object DateTimeUtils {
   // TODO move this to a LocalDateTime
   def fromValueOrNow(value: Option[String]): DateTime = Try{new DateTime(value.get)}.getOrElse(DateTime.now)
 
-  def timeUntilNextInterval(interval: FiniteDuration, now: ZonedDateTime = now()): FiniteDuration = {
+  def timeUntilNextInterval(interval: FiniteDuration, now: ZonedDateTime = DateTimeUtils.now()): FiniteDuration = {
```
Comment (bryophyta) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/DateTimeUtils.scala:23]: 👍 👍

File: cropper/app/lib/Crops.scala [hunk 2]
```diff
@@ -60,7 +60,7 @@ class Crops(config: CropperConfig, store: CropStore, imageOperations: ImageOpera
   def createCrops(sourceFile: File, dimensionList: List[Dimensions], apiImage: SourceImage, crop: Crop, cropType: MimeType)(implicit logMarker: LogMarker): Future[List[Asset]] = {
     logger.info(logMarker, s"creating crops for ${apiImage.id}")
 
-    Future.sequence[Asset, List](dimensionList.map { dimensions =>
```
Comment (bryophyta) [inline cropper/app/lib/Crops.scala:63]: i'm not sure i understand why this type annotation has been removed -- is it just because it's not necessary?

**Review by bryophyta — COMMENTED** [review]
I've read through all the commits so far (26 at the time of writing, with the most recent being `Add explicit type annotations for implicit definitions`) and it all looks good to me 👍 

I won't review as such yet, because the PR is still marked as draft.

File: cropper/app/lib/Crops.scala (hunk 2 above)
Comment (andrew-nowak) [reply cropper/app/lib/Crops.scala:63]: yes, it's not necessary and the compiler can infer the correct type. It's motivated by a change in the signature, going from [2 generic arguments in 2.12](https://www.scala-lang.org/api/2.12.20/scala/concurrent/Future$.html#sequence[A,M[X]%3C:TraversableOnce[X]](in:M[scala.concurrent.Future[A]])(implicitcbf:scala.collection.generic.CanBuildFrom[M[scala.concurrent.Future[A]],A,M[A]],implicitexecutor:scala.concurrent.ExecutionContext):scala.concurrent.Future[M[A]]) to [3 in 2.13](https://www.scala-lang.org/api/2.13.15/scala/concurrent/Future$.html#sequence[A,CC[X]%3C:IterableOnce[X],To](in:CC[scala.concurrent.Future[A]])(implicitbf:scala.collection.BuildFrom[CC[scala.concurrent.Future[A]],A,To],implicitexecutor:scala.concurrent.ExecutionContext):scala.concurrent.Future[To])

File: build.sbt [hunk 3]
```diff
@@ -74,6 +75,9 @@ val maybeBBCLib: Option[sbt.ProjectReference] = if(bbcBuildProcess) Some(bbcProj
 
 lazy val commonLib = project("common-lib").settings(
   libraryDependencies ++= Seq(
+    // FIXME - added temporarily to assist code compatible with scala 2.12 and 2.13
+    // remove ASAP after completing 2.13 upgrade!!!
+    "org.scala-lang.modules" %% "scala-collection-compat" % "2.12.0",
```
[the remark this answers is not in the document]
Comment (andrew-nowak) [reply build.sbt:80]: great cross-PR catch! added in e94c08e7eff21ff82d868f9b36f19ae60f9803b5

**Pull request merged**